tests/integration/targets/ndo_l3out_template/tasks/main.yml

```diff
@@ -31,6 +31,25 @@
   when: version.current.version is version('4.2', '>')
   block:
     # Setup Part
+    - name: Ensure ansible_test site exist
+      cisco.mso.mso_site:
+        <<: *mso_info
+        site: '{{ mso_site | default("ansible_test") }}'
+        state: query
+      register: ansible_test_site
+
+    - name: Ensure ansible_test tenant exist
+      cisco.mso.mso_tenant:
+        <<: *mso_info
+        tenant: '{{ ansible_tenant | default("ansible_test") }}'
+        users:
+          - "{{ mso_username }}"
+        sites:
+          - '{{ mso_site | default("ansible_test") }}'
+        state: present
+      register: ansible_test_tenant
+      when: ansible_test_site.current.common.name == 'ansible_test'
+
```

Comment:
Why moving this "present" requirements for the task before the other cleanup tasks?
We have been having all the cleanup then the pre-reqs so it is clearer and this invert them but I don't understand what it brings?

Reply:
Reverted the changes.